### PR TITLE
added @trigger.dev/react on CLI init and the necessary documentation

### LINKED ISSUE #330: [TRI-1018] Add the `@trigger.dev/react` package when doing `@trigger.dev/cli init`
We should also add the a bit about setting up React hooks from [this guide](https://trigger.dev/docs/documentation/guides/react-hooks) to the Next.js quickstart guide

<sub>[TRI-1018](https://linear.app/triggerdotdev/issue/TRI-1018/add-the-triggerdevreact-package-when-doing-triggerdevcli-init)</sub>

### LINKED ISSUE #331: [TRI-1029] `@trigger.dev/react` doesn't work out of the box with Next.js Page Router: Cannot use import statement outside a module Error
When using `@trigger.dev/react` in Next.js Page Router, importing will cause an error because Page Router modules aren't automatically transpiled to CJS (whereas they are in App Router).

From Next 13.1 onwards, the fix is to the add `@trigger.dev/react` to the list of transpiled packages in the next.config.js file:

```js
const nextConfig = {
  transpilePackages: ["@trigger.dev/react"],
};

module.exports = nextConfig;
```

In Next versions below 13.1, the [next-transpile-modules](https://www.npmjs.com/package/next-transpile-modules) package should be used in `next.config.js`:

```js next.config.js
const withTM = require('next-transpile-modules')(['@trigger.dev/react']); // pass the modules you would like to see transpiled

module.exports = withTM({});
```

We should do the following:

- [ ] Look into why Next.js is using the ESM module exported by `@trigger.dev/react` instead of the CJS
- [ ] If nothing can be done about the above issue, then we should make sure `@trigger.dev/cli init` correctly handles this (whenever the `@trigger.dev/react` stuff is added to the `init` command (see https://github.com/triggerdotdev/trigger.dev/issues/330)
- [ ] Add to the React Hooks docs (and the Next.js quickstart) info about this

<sub>[TRI-1029](https://linear.app/triggerdotdev/issue/TRI-1029/triggerdevreact-doesnt-work-out-of-the-box-with-nextjs-page-router)</sub>

### PR DESCRIPTION
**Pull request by Chigala**
## Changes
- Added trigger.dev/react package to the Next.js project when `@trigger.dev/cli init` is used. 
-  updated the next.js guide to include the use of react hooks

fixes #330 
temporary fix #331

### CODE REVIEW

File: packages/cli/src/commands/init.ts [hunk 1]
```diff
@@ -726,3 +740,16 @@ async function setupEnvironmentVariable(
     logger.success(`✅ Added ${variableName}=${renderer(value)} to ${fileName}`);
   }
 }
+
+async function detectNextJsVersion(projectPath: string): Promise<number> {
+  const packageFile = pathModule.join(projectPath, "package.json");
+  const packageData = await fs.readFile(packageFile, "utf8");
+  const packageJson = JSON.parse(packageData);
```
Comment (matt-aitken) [inline packages/cli/src/commands/init.ts]: You should use the `readPackageJson()` function. It returns a typed `PackageJson` object, or undefined if the files doesn't exist.

File: packages/cli/src/commands/init.ts [hunk 2]
```diff
@@ -47,6 +47,9 @@ export const initCommand = async (options: InitCommandOptions) => {
   // Detect if are are in a Next.js project
   const isNextJsProject = await detectNextJsProject(resolvedPath);
 
+  //Detect Next.Js version
+  const nextJsVersion = await detectNextJsVersion(resolvedPath)
+
```
Comment (matt-aitken) [inline packages/cli/src/commands/init.ts]: I think it'd be better if this code was after the check for if it's a Next.js project, so moved to line 60.

File: packages/cli/src/commands/init.ts [hunk 3]
```diff
@@ -145,15 +149,26 @@ export const initCommand = async (options: InitCommandOptions) => {
...
   const projectUrl = `${options.triggerUrl}/orgs/${authorizedKey.organization.slug}/projects/${authorizedKey.project.slug}`;
 
   logger.success(`✅ Successfully initialized Trigger.dev!`);
 
+
+if (nextJsDir === "pages") {
+  const message = `Detected you are using Next.js Page Router version ${nextJsVersion}, check ${
+    nextJsVersion > 13.1
+      ? "out this article on our docs on how to complete the setup. https://trigger.dev/docs/documentation/quickstarts/nextjs"
+      : "out this article on our docs on how to complete the setup. https://trigger.dev/docs/documentation/quickstarts/nextjs"
```
Comment (matt-aitken) [inline packages/cli/src/commands/init.ts]: Should these two lines say different things?

**Review by matt-aitken — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/init.ts (hunk 3 above)
Comment (Chigala) [reply packages/cli/src/commands/init.ts]: it shouldn't actually, I thought there was a way to link the  particular part of the docs with the guide for the changes to be made for the particular Next.js version with mintlify.

File: packages/cli/src/commands/init.ts (hunk 1 above)
Comment (Chigala) [reply packages/cli/src/commands/init.ts]: I'll do this instead

File: packages/cli/src/commands/init.ts (hunk 3 above)
Comment (matt-aitken) [reply packages/cli/src/commands/init.ts]: In which case would it better to just do this:

```
const message = `Detected you are using Next.js Page Router version ${nextJsVersion}, check out this article on our docs on how to complete the setup. https://trigger.dev/docs/documentation/quickstarts/nextjs`
```

Comment (Chigala) [reply packages/cli/src/commands/init.ts]: I'll go with this instead, I couldn't find a way to link to a particular part of a doc on mintlify.

File: docs/documentation/quickstarts/nextjs.mdx [hunk 4]
```diff
@@ -67,6 +67,182 @@ It will ask you a few questions
...
+The [TriggerProvider](/sdk/react/triggerprovider) component is a React Context Provider that will make the Trigger API client available to all child components.
+
+Generally you'll want to add this to the root of your app, so that it's available everywhere. However, you can add it lower in the hierarchy but it must be above any of the hooks.
+
+```tsx app/layout.tsx
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <body className={inter.className}>
+        <TriggerProvider publicApiKey={process.env.NEXT_PUBLIC_TRIGGER_API_KEY!}>
```
Comment (ericallam) [inline docs/documentation/quickstarts/nextjs.mdx:82]: We need to add this environment variable during `init` as well. You'll need to add `pkApiKey` to the `WhoamiResponseSchema` here: https://github.com/triggerdotdev/trigger.dev/blob/302bd02ff16bdba5dce811e93c605d0f832c7920/packages/cli/src/utils/triggerApi.ts#L30

The `/api/v1/whoami` endpoint already returns the `pkApiKey` so no changes will need to be made on the API side.

**Review by ericallam — CHANGES_REQUESTED** [review]

File: docs/documentation/quickstarts/nextjs.mdx (hunk 4 above)
Comment (Chigala) [reply docs/documentation/quickstarts/nextjs.mdx:82]: I'll fix this ASAP.

File: packages/cli/src/commands/init.ts [hunk 5]
```diff
@@ -671,6 +684,20 @@ export * from "./examples"
 }
 
 async function setupEnvironmentVariables(path: string, options: ResolvedOptions) {
+  if (options.apiKey && options.apiUrl) {
+    const apiClient = new TriggerApi(options.apiKey, options.apiUrl);
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts]: `whoami` is already being called during init, is it possible to do this without calling it twice?

Comment (Chigala) [reply packages/cli/src/commands/init.ts]: Sorry, I didn't see it was already being called.

File: packages/cli/src/commands/init.ts [hunk 6]
```diff
@@ -145,15 +148,26 @@ export const initCommand = async (options: InitCommandOptions) => {
...
+  authorizedKey: WhoamiResponse,
+  nextJsVersion: number,
+  nextJsDir: "pages" | "app"
+) {
   const projectUrl = `${options.triggerUrl}/orgs/${authorizedKey.organization.slug}/projects/${authorizedKey.project.slug}`;
 
   logger.success(`✅ Successfully initialized Trigger.dev!`);
 
+  if (nextJsDir === "pages") {
+    const message = `Detected you are using Next.js Page Router version ${nextJsVersion}, check out this article on our docs on how to complete the setup. https://trigger.dev/docs/documentation/quickstarts/nextjs`;
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts:166]: I'm not sure what this is meant to be for? Is this for doing the transpile module stuff? It doesn't look like we actually have anything in our docs referencing how to finish setting up the Page Router version

File: packages/cli/src/commands/init.ts [hunk 7]
```diff
@@ -670,7 +683,18 @@ export * from "./examples"
   }
 }
 
-async function setupEnvironmentVariables(path: string, options: ResolvedOptions) {
+async function setupEnvironmentVariables(path: string, options: ResolvedOptions, authorizedKey: WhoamiResponse) {
+  if (options.apiKey && options.apiUrl) {
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts:687]: Do we need this conditional here?

File: packages/cli/src/commands/init.ts [hunk 8]
```diff
@@ -726,3 +750,21 @@ async function setupEnvironmentVariable(
     logger.success(`✅ Added ${variableName}=${renderer(value)} to ${fileName}`);
   }
 }
+
+async function detectNextJsVersion(projectPath: string): Promise<number> {
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts:754]: Why is this function being added here?

**Review by ericallam — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/init.ts (hunk 6 above)
Comment (Chigala) [reply packages/cli/src/commands/init.ts:166]: I updated the docs in the first commit to show how to do that

File: packages/cli/src/commands/init.ts (hunk 8 above)
Comment (Chigala) [reply packages/cli/src/commands/init.ts:754]: For the log message that'll have the link to the blog where there's a guide on how to complete @trigger.dev/react set up.

**Comment by ericallam** [comment]
@matt-aitken is currently doing a big refactor on the CLI to account for multi-framework support. Matt could you incorporate some of these changes into the work that you are doing? I've going to close this for now.

**Pull request closed without merging**